lib/ansible/modules/cloud/amazon/ec2_vpc_route_table.py

```diff
@@ -294,7 +294,7 @@ def find_subnets(connection, module, vpc_id, identified_subnets):
             module.fail_json_aws(e, msg="Couldn't find subnet with names %s" % subnet_names)
 
         for name in subnet_names:
-            matching_count = len([1 for s in subnets_by_name if s.tags.get('Name') == name])
+            matching_count = len([1 for s in subnets_by_name if s['Tags'][0].get('Value') == name])
```

Comment:
How could this happen? I clearly stuffed this bit up in the boto3 conversion and no-one noticed. 

I don't think this improvement will work unless the first tag is always the Name tag, which seems unlikely.

This should probably be 

```
matching_count = len([1 for s in subnets_by_name for t in s.get('Tags', []) if t['Name'] == 'Name' and t['Value'] == name])
```

Could you add a test for this situation into test/integration/targets/ec2_vpc_route_table/tasks/main.yml (tests would hopefully have caught both the original error and the flaw in this solution)